### PR TITLE
target selection only for driving navigation

### LINKED ISSUE #14344: "map" navigation shows popup with parking coordinates
In todays nightly pressing the compass icon which in my config is mapped to "map" (default navigation) opens a popup showing the cache and its parking coordinates instead of just drawing the line to the cache.

Showing where a cache is is THE main function of cgeo and this popup, requiring an additional selection - each time the button is pressed, which can be "very often" if you switch between listing and map - adds useless overhead to this.
If i want to draw a line to the parking coordinates i can go to the waypoints and select parking there.

### PR DESCRIPTION
**Pull request by ztNFny**
fixes #14344 by offering target selection only for those navigation apps that are explictily targeted towards driving

Apps with explicit "Driving" navigation mode in cgeo:
- Google Navigation (Driving)
- Sygic (Driving)

Other navigation apps not implementing this right now:
- Maps.me
- OruxMaps
- OsmAnd
- Locus

If we want to implement it for those other apps I suggest adding a second profile for them to explicitly set driving mode, then implement it for that.

### CODE REVIEW

File: main/src/main/java/cgeo/geocaching/apps/navi/AbstractPointNavigationApp.java [hunk 1]
```diff
@@ -36,24 +36,30 @@ protected AbstractPointNavigationApp(@NonNull final String name, @Nullable final
...
+        assert coords != null; // asserted by caller
+        navigate(context, coords);
+    }
+
+    @Override
+    public void navigateWithTargetSelector(@NonNull final Context context, @NonNull final Geocache cache) {
         final ArrayList<IWaypoint> targets = new ArrayList<>();
         targets.add(cache);
         for (final Waypoint waypoint : cache.getWaypoints()) {
-            final Geopoint coords = waypoint.getCoords();
```
Comment (moving-bits) [inline main/src/main/java/cgeo/geocaching/apps/navi/AbstractPointNavigationApp.java:50]: By changing the next two lines you also revert #14339 (which is not related to the reverted #14320)

File: main/src/main/java/cgeo/geocaching/apps/navi/AbstractPointNavigationApp.java [hunk 2]
```diff
@@ -36,24 +36,30 @@ protected AbstractPointNavigationApp(@NonNull final String name, @Nullable final
     }
 
     @Override
-    public void navigateWithoutSelector(@NonNull final Context context, @NonNull final Geocache cache) {
```
Comment (moving-bits) [inline main/src/main/java/cgeo/geocaching/apps/navi/AbstractPointNavigationApp.java:39]: By reversing the logic from `navigateWithoutSelector` back to `navigate` navigation you'll end up in a loop if you select the geocache (not one of its waypoints) as navigation target.

**Review by moving-bits — CHANGES_REQUESTED** [review]

File: main/src/main/java/cgeo/geocaching/apps/navi/AbstractPointNavigationApp.java (hunk 1 above)
Comment (ztNFny) [reply main/src/main/java/cgeo/geocaching/apps/navi/AbstractPointNavigationApp.java]: re-implemented

**Comment by ztNFny** [comment]
retest this please

**Review by moving-bits — APPROVED** [review]
LGTM, thanks for fixing the issues

**Pull request merged**